waspls/watch

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# Rebuilds project when a source file changes
+
+inotifywait -m -e close_write -r src/ exe/ waspls.cabal |
+  while read directory action file; do
+    printf "\033[33mFile changed. Starting recompile...\033[0m\n"
+    cabal build
+    printf "\033[1;32mFinished\033[0m\n"
+  done
```

Comment:
What is this, why do we need it?

Reply:
It seems very specific so that is why I would avoid committing if not everybody needs it.
But it is short and does it's job so I guess why not.
I am surprised cabal doesn't have this functionality!
What about ghcid, I thought `ghcid` is the solution for this?
What about LSP, why that doesn't work for you?

Reply:
LSP/ghcid work fine for, but don't do exactly what this file does. The purpose of this script is to produce a `waspls` exe, which I can test in the vscode extension. Both Haskell LSP and ghcid don't actually produce an executable.

And I agree, it's strange cabal doesn't have something like `cabal watch` to do this.